exonum-java-binding-core/rust/ejb-app/src/main.rs

```diff
@@ -1,12 +1,67 @@
+extern crate exonum_btc_anchoring;
 extern crate exonum_configuration;
 extern crate java_bindings;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
 
+use exonum_btc_anchoring::ServiceFactory as BtcAnchoringServiceFactory;
 use exonum_configuration::ServiceFactory as ConfigurationServiceFactory;
-use java_bindings::exonum::helpers::fabric;
+use java_bindings::exonum::helpers::config::ConfigFile;
+use java_bindings::exonum::helpers::fabric::{self, ServiceFactory};
+
+use std::collections::{HashMap, HashSet};
+
+const PATH_TO_SERVICES_TO_ENABLE: &str = "ejb_app_services.toml";
+const CONFIGURATION_SERVICE: &str = "configuration";
+const BTC_ANCHORING_SERVICE: &str = "btc-anchoring";
+const EJB_SERVICE: &str = "ejb-service";
+
+#[derive(Serialize, Deserialize)]
+struct ServicesToEnable {
+    services: HashSet<String>,
+}
+
+fn service_factories() -> HashMap<String, Box<ServiceFactory + 'static>> {
+    let mut service_factories = HashMap::new();
+    service_factories.insert(
+        CONFIGURATION_SERVICE.to_owned(),
+        Box::new(ConfigurationServiceFactory) as Box<ServiceFactory>,
+    );
+    service_factories.insert(
+        BTC_ANCHORING_SERVICE.to_owned(),
+        Box::new(BtcAnchoringServiceFactory) as Box<ServiceFactory>,
+    );
+    service_factories.insert(
+        EJB_SERVICE.to_owned(),
+        Box::new(java_bindings::JavaServiceFactory) as Box<ServiceFactory>,
+    );
+    service_factories
+}
 
 fn main() {
-    let builder = fabric::NodeBuilder::new()
-        .with_service(Box::new(java_bindings::JavaServiceFactory))
-        .with_service(Box::new(ConfigurationServiceFactory));
+    let ServicesToEnable { mut services } =
+        ConfigFile::load(PATH_TO_SERVICES_TO_ENABLE).unwrap_or(ServicesToEnable {
+            services: {
+                let mut services = HashSet::new();
+                services.insert(CONFIGURATION_SERVICE.to_owned());
+                services
+            },
+        });
+
+    let mut service_factories = service_factories();
+
+    // Add EJB_SERVICE if it's missing
+    services.insert(EJB_SERVICE.to_owned());
+
+    let mut builder = fabric::NodeBuilder::new();
+    for service_name in &services {
+        match service_factories.remove(service_name) {
```

Comment:
I think of this map of factories as an immutable map, but see it's not the case. I also see a new one is created when `service_factories` is called, so it’s OK in terms of correctness to remove the factories from it. This process looks a little complicated. Is there a good reason to not use an immutable map?

Reply:
Immutable map will give only references to values to me which is not enough - I need owned values. Well, technically, we can use closures stored in HashMap. Will try and look at complexity. 
To be honest, the current solution is much more complicated then first attempt with simple `if...else`

Reply:
> To be honest, the current solution is much more complicated

Yes, it is more complex. Rust complicated things :-) If/else is repetitive and error prone (you have to test each branch basically), but if a solution with a map results in complex Rust code, we can go back to it.